src/mam4xx/ndrop.hpp

```diff
@@ -47,5 +47,51 @@ void get_aer_num(const Diagnostics &diags, const Prognostics &progs,
   naerosol[mode_idx] = max(naerosol[mode_idx], vaerosol * num2vol_ratio_max);
   naerosol[mode_idx] = min(naerosol[mode_idx], vaerosol * num2vol_ratio_min);
 }
+
+KOKKOS_INLINE_FUNCTION
+void explmix(
+    int pver,     // number of levels
+    Real q[4],    // number / mass mixing ratio to be updated [# or kg / kg]
+    Real src[4],  // source due to activation/nucleation [# or kg / (kg-s)]
+    Real ekkp[4], // zn*zs*density*diffusivity (kg/m3 m2/s) at interface [/s];
+                  // below layer k  (k,k+1 interface)
+    Real ekkm[4], // zn*zs*density*diffusivity (kg/m3 m2/s) at interface [/s];
+                  // above layer k  (k,k+1 interface)
+    Real overlapp[4], // cloud overlap below [fraction]
+    Real overlapm[4], // cloud overlap above [fraction]
+    Real qold[4], // number / mass mixing ratio from previous time step [# or kg
+                  // / kg]
+    Real dt,      // time step [s]
+    bool is_unact,  // true if this is an unactivated species
+    Real qactold[4] // optional: number / mass mixing ratio of ACTIVATED species
+                    // from previous step *** this should only be present if the
+                    // current species is unactivated number/sfc/mass
+) {
+
+  int top_lev = 0; //??
```

Comment:
I assume we start indexing at 0?

Reply:
Yes, that's correct. And the indexing goes from the top of the atmosphere to the surface of the earth, for reasons related to the "dycore" (dynamical core), which moves things around between columns.